client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-select-checkbox/index.tsx

```diff
@@ -17,15 +17,15 @@ export default function SiteSelectCheckbox( { item, siteError, isLargeScreen }:
 	);
 
 	function handleCheckboxClick() {
-		item.onSelect( item );
+		item.onSelect?.();
 		recordEvent( item.isSelected ? 'site_unselected' : 'site_selected' );
 	}
 
 	return (
 		<span className="site-select-checkbox">
 			<FormInputCheckbox
 				className="disable-card-expand"
-				id={ item.blog_id }
+				id={ `${ item.site.value.blog_id }` }
```

Comment:
`id` needs to be a string. It was being sent as undefined earlier. We are now using `item.site.value.blog_id`. Thanks to TS